### ISSUE TITLE
Issues with fluent-plugin-splunk-hec and latest v1.16-full

### ISSUE THREAD

**Issue by ElfoLiNk** [kube-logging/fluentd-images#107]
Fluentd is crashing with the following error:

```
/usr/local/lib/ruby/3.3.0/rubygems/specification.rb:2307:in `check_version_conflict': can't activate timeout-0.4.0, already activated timeout-0.4.1 (Gem::LoadError)
        from /usr/local/lib/ruby/3.3.0/rubygems/specification.rb:1446:in `activate'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:98:in `each'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:98:in `block in require'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:39:in `synchronize'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:39:in `require'
        from /usr/local/bundle/gems/connection_pool-2.4.1/lib/connection_pool.rb:1:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from /usr/local/bundle/gems/net-http-persistent-4.0.2/lib/net/http/persistent.rb:4:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from /usr/local/bundle/gems/fluent-plugin-splunk-hec-1.3.3/lib/fluent/plugin/out_splunk_hec.rb:11:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/registry.rb:103:in `block in search'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/registry.rb:100:in `each'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/registry.rb:100:in `search'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/registry.rb:44:in `lookup'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/plugin.rb:169:in `new_impl'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/plugin.rb:114:in `new_output'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/agent.rb:130:in `add_match'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/agent.rb:74:in `block in configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/agent.rb:64:in `each'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/agent.rb:64:in `configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/label.rb:31:in `configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/root_agent.rb:146:in `block in configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/root_agent.rb:146:in `each'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/root_agent.rb:146:in `configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/engine.rb:105:in `configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/engine.rb:80:in `run_configure'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/supervisor.rb:571:in `run_supervisor'
        from /usr/local/bundle/gems/fluentd-1.16.3/lib/fluent/command/fluentd.rb:352:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
        from /usr/local/bundle/gems/fluentd-1.16.3/bin/fluentd:15:in `<top (required)>'
        from /usr/local/bundle/bin/fluentd:25:in `load'
        from /usr/local/bundle/bin/fluentd:25:in `<main>'
```

**Comment by pepov** [comment]
@ElfoLiNk thanks for reporting! Could you please also provide a very simple config to reproduce this? Can this be reproduced without an actual splunk account as well?

**Comment by ElfoLiNk** [comment]
Hi @pepov you can follow https://kube-logging.dev/docs/examples/splunk/

**Comment by pepov** [comment]
@ElfoLiNk I haven't been able to get back to reproducing this, but I saw the plugin is now EOL and we are not aware of any maintained alternatives. Regardless of this we still intend to fix this issue, although with low priority currently. Anyone feel free to jump in and submit a PR with a fix.

Also, for anyone been hit by this, the syslog-ng aggregator can be an alternative solution if you can consider rewriting your flows into syslog-ng flows:
https://kube-logging.dev/docs/configuration/log-routing-syslog-ng/
If anyone need help with that please hit us up on our discord https://discord.gg/z85S92pKxH

**Comment by ElfoLiNk** [comment]
@pepov i think because i have also an opensearch clusteroutput sorry I forgot to mention it

**Comment by pepov** [comment]
There are other problems as well with the latest image which uses ruby3.3 and the latest alpine, so we are going to switch back with the latest until it gets fixed

**Comment by pepov** [comment]
https://github.com/kube-logging/fluentd-images/pull/111

**Comment by pepov** [comment]
we've rolled back with the image, but the faulty one will be available under `v1.16-ruby3.3-full`

we will get back to checking this occassionally, so I leave this open

**Comment by pepov** [comment]
Created a tracking issue for the upgrade and closing this one
